app/server/lib/ActiveDoc.ts

```diff
@@ -2592,6 +2600,33 @@ export class ActiveDoc extends EventEmitter {
     }];
   }
 
+  private async _updateAttachmentFileSizesUsingFileIdent(docSession: OptDocSession,
+                                                         newFileSizesByFileIdent: Map<string, number>): Promise<void> {
+    const rowIdsToUpdate: number[] = [];
+    const newFileSizesForRows: CellValue[] = [];
+    const attachments = this.docData?.getMetaTable("_grist_Attachments").getRecords();
+    for (const attachmentRec of attachments ?? []) {
+      const newSize = newFileSizesByFileIdent.get(attachmentRec.fileIdent);
+      if (newSize) {
```

Comment:
How about checking if the size is different from the old recorded size? I believe the data engine will prune changes where a value is written to a cell that already contains that value, but seems easy to cut it out at the source here.